### ISSUE TITLE
push works locally, but build fails on CI

### ISSUE THREAD

**Issue by mustafahafidi** [aws-amplify/amplify-cli#11908]
### Amplify CLI Version

latest (live package on)

### What operating system are you using?

I am using Ubuntu (linux) locally. You provision Amazon linux in your CI

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

-

### Describe the bug

`Amplify push` successfully deploys the backend.
Branch based builds worked, until I created a custom resource (which succeeds to deploy locally).
Now CI build fails with:
```
                                 # Starting phase: build
2023-02-06T19:12:40.755Z [INFO]: ## Running amplify pull to generate aws-exports.js file for frontend
2023-02-06T19:12:43.007Z [INFO]: [0mAmplify AppID found: d2e16t1yf30k3p. Amplify App name is: abiotweb[0m
2023-02-06T19:12:43.055Z [INFO]: [0mBackend environment dev found in Amplify Console app: abiotweb[0m
2023-02-06T19:12:47.037Z [WARNING]: - Fetching updates to backend environment: dev from the cloud.
2023-02-06T19:12:50.508Z [WARNING]: - Building resource api/AdminQueries
2023-02-06T19:12:50.539Z [WARNING]: - Building resource api/abiotweb
2023-02-06T19:12:51.202Z [INFO]: ⚠️ WARNING: owners may reassign ownership for the following model(s) and role(s): Consortium: [id]. If this is not intentional, you may want to apply field-level authorization rules to these fields. To read more: https://docs.amplify.aws/cli/graphql/authorization-rules/#per-user--owner-based-data-access.
2023-02-06T19:12:52.142Z [INFO]: ✅ GraphQL schema compiled successfully.
                                 Edit your schema at /codebuild/output/src644533651/src/abiot-agriculture/amplify/backend/api/abiotweb/schema.graphql or place .graphql files in a directory at /codebuild/output/src644533651/src/abiot-agriculture/amplify/backend/api/abiotweb/schema
2023-02-06T19:12:52.143Z [WARNING]: - Building resource auth/abiotwebAuthCFNResource
2023-02-06T19:12:52.216Z [WARNING]: - Building resource auth/userPoolGroups
2023-02-06T19:12:52.315Z [WARNING]: - Building resource custom/IoTInboundProcessingRule
2023-02-06T19:12:52.315Z [WARNING]: - Building custom resources
2023-02-06T19:12:52.356Z [WARNING]: ✖ There was an error initializing your environment.
2023-02-06T19:12:52.579Z [INFO]: 🛑 There was an error building the custom resources
2023-02-06T19:12:52.579Z [INFO]: No package manager found. Please install npm or yarn to compile overrides for this project.
                                 Resolution: There may be errors in your custom resource file. If so, fix the errors and try again.
                                 Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/
2023-02-06T19:12:52.589Z [INFO]: 
2023-02-06T19:12:52.589Z [INFO]: Session Identifier: 1063b223-a32c-4820-a40d-fe6f25a4b8e8
2023-02-06T19:12:52.610Z [ERROR]: !!! Build failed
2023-02-06T19:12:52.610Z [ERROR]: !!! Non-Zero Exit Code detected
2023-02-06T19:12:52.611Z [INFO]: # Starting environment caching...
2023-02-06T19:12:52.611Z [INFO]: # Environment caching completed
Terminating logging...
```

### Expected behavior

expect it to succeed

### Reproduction steps

created custom resource using cdk.
```
…:~/c/a/a/b/c/abiotIotInboundRule]─[07:23:00 PM]─[G:main=]
╰─>$ ls
build/        node_modules/  package-lock.json  yarn.lock
cdk-stack.ts  package.json   tsconfig.json
…:~/c/a/a/b/c/abiotIotInboundRule]─[07:23:00 PM]─[G:main=]
╰─>$ pwd
/home/mustafa/coding/abiot-agriculture/amplify/backend/custom/abiotIotInboundRule
```

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by josefaidt** [comment]
Hey @mustafahafidi :wave: thanks for raising this! I was not able to reproduce this by adding a custom resource and deploying with Hosting's CI/CD. What stood out to me is you have two separate lockfiles (one for npm and the other for yarn), though even adding this to my setup was not able to cause the error.

The error was coming from here https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-category-custom/src/utils/build-custom-resources.ts#L66 which is resolving the package manager from here https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-cli-core/src/utils/packageManager.ts#L32-L36

Are there any customizations applied to your project's `amplify.yml`?

**Comment by mustafahafidi** [comment]
Not really, I am using the default amplify build settings (i.e. had no amplify.yml file); now as workaround I am forced to disable backend deploys, and perform those manually.

**Comment by josefaidt** [comment]
Hey @mustafahafidi thanks for clarifying! It appears this issue is similar to an active bug where it is also failing to build the custom resource in Amplify Hosting's CI/CD https://github.com/aws-amplify/amplify-cli/issues/10723

I'll mark this as a duplicate bug and later close this in favor of tracking the existing item, however I'd like to understand how you are encountering this issue in order to provide an effective workaround and get you unblocked.

I have tried creating a new custom resource and pushing to git (without running `amplify push`) first to no avail. Do you see a mismatch in resource names between your `amplify/backend/backend-config.json` file and your directory names? 

```json
"custom": {
    "customResource0f0430f0": {
      "providerPlugin": "awscloudformation",
      "service": "customCDK"
    },
    "customResource9976a2a6": {
      "providerPlugin": "awscloudformation",
      "service": "customCDK"
    }
  },
```

**Comment by mustafahafidi** [comment]
No mismatch:

```
"custom": {
    "IoTInboundProcessingRule": {
      "providerPlugin": "awscloudformation",
      "service": "customCDK"
    },
    "abiotIotInboundRule": {
      "dependsOn": [
        {
          "attributes": [
            "Name",
            "Arn",
            "Region",
            "LambdaExecutionRole",
            "LambdaExecutionRoleArn"
          ],
          "category": "function",
          "resourceName": "abiotIoTInboundProcessor"
        }
      ],
      "providerPlugin": "awscloudformation",
      "service": "customCDK"
    }
  },
```
(see pwd command in above's comment)
I expect it to fail locally if it was the case.

The error message sounds like whatever process you guys do behind the scenes in that `amplifyPush --simple` is making it fail to find npm/yarn?

**Comment by mustafahafidi** [comment]
Also I feel like this isn't a duplicate of https://github.com/aws-amplify/amplify-cli/issues/10723

**Comment by josefaidt** [comment]
Hey @mustafahafidi thanks for clarifying! I attempted to reproduce this again with a resource named with a capital first letter (shown here and #10723) as well as creating a dependency on another Amplify resource to no avail. Does the casing of the directory names match their respective resource names?

**Comment by josefaidt** [comment]
Hey @mustafahafidi I wanted to follow-up and see if you are you still experiencing this?

**Comment by josefaidt** [comment]
Closing due to inactivity
